monkey/monkey_island/cc/services/authentication_service.py

```diff
@@ -21,17 +21,21 @@ def __init__(
         self._repository_encryptor = repository_encryptor
         self._island_event_queue = island_event_queue
 
-    def register_new_user(self, username: str, password: str):
+    def needs_registration(self) -> bool:
         """
-        Registers a new user on the Island, then resets the encryptor and database
+        Checks if a user is already registered on the Island
 
-        :param username: Username to register
-        :param password: Password to register
-        :raises InvalidRegistrationCredentialsError: If username or password is empty
+        :return: Whether registration is required on the Island
         """
-        if not username or not password:
-            raise InvalidRegistrationCredentialsError("Username or password can not be empty.")
+        return not User.objects.first()
 
+    def reset_island(self, username: str, password: str):
```

Comment:
I don't think we need this method. I think publishing those events are reusable (resetting the island could happen not only on registration we could decide that for convenience sake we want to provide reset HTTP endpoint for example). Unlocking repository encryptor is a different action, so I think it would be best to split it up

Reply:
I agree with that just didn't had time to do it yesterday

Reply:
They are not needed anywhere right now. We do have some reset HTTP endpoints like `ClearSimulationData`. It is not adding any value and if needed we can easily split them up.